### PR TITLE
Poll for status updates and linode additions/deletions

### LINKED ISSUE #695: Poll for updates on a per-page basis

### PR DESCRIPTION
**Pull request by eatonphil**
Closes #695

### CODE REVIEW

File: src/api/configs/datacenters.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import {
 export const config = genConfig({
   plural: 'datacenters',
   singular: 'datacenter',
-  crossUserCache: true,
+  localstorageCacheable: true,
```
Comment (briansteffens) [inline src/api/configs/datacenters.js]: > local**s**torageCacheable

*twitch*

File: src/api/gen.js [hunk 2]
```diff
@@ -142,18 +130,26 @@ export function genReducer(_config) {
...
-    // data is not immutable.
-
-    // Furthermore, this does not support invalidating invidual top-level
-    // states like a single Linode.
-    return {
-      ...state,
-      ...genDefaultState(config, action.partial && state),
-    };
+    let newState = { ...state };
+    if (action.partial) {
```
Comment (briansteffens) [inline src/api/gen.js]: What does partial mean?

Comment (eatonphil) [inline src/api/gen.js]: I comment about it in a few places. Wasn't sure if I should be saying the same thing everywhere.

Comment (eatonphil) [inline src/api/gen.js]: Actually, it's commented on in the line below this.

File: src/linodes/linode/layouts/IndexPage.js [hunk 3]
```diff
@@ -47,8 +47,8 @@ export class IndexPage extends Component {
     const { linodeId } = newParams;
 
     try {
-      await store.dispatch(linodes.one(linodeId));
-      await store.dispatch(linodes.configs.all(linodeId));
+      await store.dispatch(linodes.one([linodeId]));
```
Comment (eatonphil) [inline src/linodes/linode/layouts/IndexPage.js]: Some of these functions take an array explicitly (like this one now) and some take them implicitly (using spread args). In both cases it does not refer to multiple ids to fetch. It refers to the path of ids that get you to the resource you're talking about. For instance, configs.one([linodeId, configId]) gets you the specific config on the specific linode. This API is obviously debatable -- I didn't change it here. But I did switch off spread args when there potentially other flags to be sent in the call.

File: src/api/gen.js (hunk 2 above)
Comment (briansteffens) [inline src/api/gen.js]: Ah gotcha

File: src/linodes/linode/layouts/IndexPage.js (hunk 3 above)
Comment (briansteffens) [inline src/linodes/linode/layouts/IndexPage.js]: Ahh ok that makes sense

File: src/api/genThunks.js [hunk 4]
```diff
@@ -60,77 +89,110 @@ function genThunkOne(config, actions) {
   };
 }
 
+/*
+ * This function fetches a single page. If the cache becomes invalidated
+ * (a.k.a. number of total results changes), all pages are fetched. Otherwise
+ * the number of pages fetched (1) and the page json are returned. All returned
+ * results are optionally filtered so only certain fields (or none) are updated.
+ */
```
Comment (eatonphil) [inline src/api/genThunks.js]: Whoops, I think this comment is out of date. It definitely no longer tries to refetch all pages.

**Comment by eatonphil** [comment]
thanks!

**Pull request merged**